experiment/compatibility-versions/e2e-two-steps-upgrade.sh

```diff
@@ -109,24 +109,36 @@ main() {
     exit $res
   fi
 
-  EMULATED_VERSION_UPGRADE_SCRIPT="${EMULATED_VERSION_UPGRADE_SCRIPT:-${PWD}/../test-infra/experiment/compatibility-versions/emulated-version-upgrade.sh}"
-  echo "Upgrading cluster with ${EMULATED_VERSION_UPGRADE_SCRIPT}"
-  "${EMULATED_VERSION_UPGRADE_SCRIPT}" | tee "${ARTIFACTS}/emulated-upgrade-output.txt"
+  # after first step of upgrading binary version without bumping emulated version, verify all tests from the previous version pass
 
   # Clone the previous versions Kubernetes release branch
   # TODO(aaron-prindle) extend the branches to test from n-1 -> n-1..3 as more k8s releases are done that support compatibility versions
-  export RELEASE_BRANCH="release-${CURRENT_VERSION}"
```

Comment:
This part just seems wrong, I'm surprised we haven't seen issues with it before. Thanks for the fix! Reminds me that I should update https://github.com/kubernetes/test-infra/pull/34930 so that we can get away from building these test binaries altogether.